src/Raku/ast/package.rakumod

```diff
@@ -125,7 +125,7 @@ class RakuAST::Package
             # We always install it as a lexical symbol.
             $resolver.current-scope.merge-generated-lexical-declaration:
                 RakuAST::Declaration::LexicalPackage.new:
-                    :lexical-name($name.canonicalize),
+                    :lexical-name($final),
```

Comment:
‘$final’ only has a value if ‘$name.is-identifier’, is that correct?

Reply:
IIUC, no. `$final` is also set in the else block below: `$final := nqp::pop(@parts).name;`.